packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -449,210 +528,170 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
-                fileAppendChunkSize,
-                exchangeRateInCents,
-              );
-
               try {
                 for (deploymentCounts = 0; deploymentCounts < 50; deploymentCounts++) {
-                  await deployContract(largeContractJson, aliasAccount.wallet);
+                  const tx = await deployContract(largeContractJson, aliasAccount.wallet);
+                  await tx.waitForDeployment();
+
+                  expectedTxCost ||= await getExpectedCostOfLastLargeTx(largeContractJson.bytecode);
+                  await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts + 1, expectedTxCost);
                 }
                 expect.fail(`Expected an error but nothing was thrown`);
               } catch (e: any) {
+                logger.error(e.message);
                 expect(e.message).to.contain(predefined.HBAR_RATE_LIMIT_EXCEEDED.message);
-                const largeContractDeploymentCost = await getExpectedCostOfLastLargeTx(deployedTransaction.data);
-                const expectedAmountOfDeployments = Math.floor(maxSpendingLimit / largeContractDeploymentCost);
+                const expectedAmountOfDeployments = Math.floor(maxSpendingLimit / expectedTxCost);
                 expect(deploymentCounts).to.eq(expectedAmountOfDeployments);
 
-                // awaiting for HBAR limiter to finish updating expenses in the background
-                await Utils.wait(6000);
-
-                const spendingPlanAssociated = await hbarSpendingPlanRepository.findByIdWithDetails(
-                  hbarSpendingPlan.id,
-                  requestDetails,
-                );
-                const amountSpent = spendingPlanAssociated.amountSpent;
+                const amountSpent = await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts, expectedTxCost);
                 const remainingHbarsAfter = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
 
                 // Explanation:
                 // An HBAR limit check triggers the HBAR_RATE_LIMIT_EXCEED error in two scenarios:
-                //    a. if remainingHbarsBefore > maxSpendingLimit ===> (totalHbarSpentByPrivilegedPlan + estimatedTxFee) > maxSpendingLimit
-                //    b. if remainingHbarsBefore <= maxSpendingLimit ===> remainingBudget < estimatedTxFee
+                //    a. if remainingHbarsBefore > maxSpendingLimit ===> (totalHbarSpentByPrivilegedPlan + expectedTxCost) > maxSpendingLimit
+                //    b. if remainingHbarsBefore <= maxSpendingLimit ===> remainingBudget < expectedTxCost
                 if (remainingHbarsBefore > maxSpendingLimit) {
-                  expect(amountSpent + estimatedTxFee).to.be.gt(maxSpendingLimit);
+                  expect(amountSpent + expectedTxCost).to.be.gt(maxSpendingLimit);
                 } else {
-                  expect(remainingHbarsAfter).to.be.lt(estimatedTxFee);
+                  expect(remainingHbarsAfter).to.be.lt(expectedTxCost);
```

Comment:
**Note for reviewers:**
The variable expectedTxCost has been renamed around just simply for the sake of consistency.